File: MS_rosterfrontend/projects/employee-dashboard/src/lib/viewmodels/shiftinfo.ts
import { ServiceTypeViewModel } from 'projects/roster/src/lib/viewmodel/servicetype-viewModel';

export interface Shiftinfo {
    id?: number;
    description?: string;
    duration?: number;
    ratio?: string;
    clientCount?: number;
    employeeCount?: number;
    locationId?: number;
    locationName?: string;
    statusName?: string;
    otherLocation?: string;
    locationType?: number;
    statusId?: number;
    isPublished?: boolean;
    startDate?: string;
    endDate?: string;
    startTime?: string;
    endTime?: string;
    startTimeString?: string;
    endTimeString?: string;
    employeeId?: EmployeeShiftInfoViewModel[];
    clientId?: [];
    shiftRepeat?: string;
    days?: number;
    reminder?: boolean;
    employeeShiftInfoViewModel?: EmployeeShiftInfoViewModel[];
    clientShiftInfoViewModel?: ClientShiftInfoViewModel[];
    serviceTypeViewModel?: ServiceTypeViewModel[];
    isActiveNight?: boolean;
    isSleepOver?: boolean;
    shiftRepeatType?: string;
    customDuration?: string;
    adminCheckoutRemark?: string;
    checkoutRemark?: string;
}

export interface EmployeeShiftInfoViewModel {
    id?: number;
    employeeId?: number;
    name?: string;
    isSleepOver?: boolean;
}

export interface ClientShiftInfoViewModel {
    id?: string;
    clientId?: number;
    name?: string;
}

File: MS_lifehealthservices/LHSAPI.Domain/Entities/ShiftInfo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace LHSAPI.Domain.Entities
{
    public class ShiftInfo : BaseEntity
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        public string Description { get; set; }

        public int ClientCount { get; set; }

        public int EmployeeCount { get; set; }

        public int? LocationId { get; set; }

        public string OtherLocation { get; set; }

        public bool IsPublished { get; set; }

        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }

        public TimeSpan StartTime { get; set; }

        public TimeSpan EndTime { get; set; }
        public DateTime StartUtcDate { get; set; }
        public DateTime EndUtcDate { get; set; }

        public bool Reminder { get; set; }

        public double Duration { get; set; }
        public int LocationType { get; set; }
        public string ShiftRepeatType { get; set; }
        public string Remark { get; set; }
    }
}
